### DISCUSSION TITLE
GRDB with swiftUI @Query question

### DISCUSSION THREAD

**Discussion by yliu342** [groue/GRDB.swift#1152] [Q&A] [answered]
Hi, I'm wanting to use GRDB in a SwiftUI app and I'm totally new to GRDB but from what I read it is really promising. I looked at the demo app and noticed there is a @Query property wrapper and I can't seem to find and doc on can anyone point me to the right direction?

Also I played with it a bit and noticed I can't use a a variable within the property wrapper (a SwiftUI limitation?). Assuming my record looks something like this:
```
struct Player: Identifiable, Equatable {
    var id: Int64?
    var name: String
    var score: Int
    var team: String
}
```
And my view want to show player in a particular `team` the `team` is passed in from the previous view so it will be dynamic. What is the best way about doing this using @Query? Or I can't use it and maybe I should go with `valueObervation` publisher?

My last question on how to best delete the table/db if I use GRDB in a SPM package that being used as part of an app? Should I simply delete the `.sqlite` file from disk or is there a better way? Assuming the connection is closed due to pool is released.

Any suggestion is very much appreciated. 
Thank you

**Comment by groue** [comment, marked as answer]
Hello @yliu342,

> there is a @query property wrapper and I can't seem to find and doc

It comes from the [GRDBQuery](http://github.com/groue/GRDBQuery) package.

> And my view want to show player in a particular team the team is passed in from the previous view so it will be dynamic.

You can get inspiration from [those lines](https://github.com/groue/GRDB.swift/blob/v5.19.0/Documentation/DemoApps/GRDBCombineDemo/GRDBCombineDemo/Views/AppView.swift#L18-L24) in the demo app.

> My last question on how to best delete the table/db if I use GRDB in a SPM package that being used as part of an app? Should I simply delete the .sqlite file from disk or is there a better way? Assuming the connection is closed due to pool is released.

I'm not sure I understand your question: deleting a table, or a whole database file, is not at all the same thing. Since deleting a table is not supposed to be difficult, I assume you ask how to delete a database file. Please see https://github.com/groue/GRDB.swift/issues/164#issuecomment-312468700

**Reply by yliu342** [reply]
Thank you for your fast reply. Sorry I should be more clear on my last question. If I understand correctly, I can't drop a table while the db connection is closed so I thought maybe delete the entire db file will be easier. Your answer to store DB files in a directory and simply delete the directory works for me as well.

Thank you again.
